include/triton/core/tritonbackend.h

```diff
@@ -594,6 +594,31 @@ TRITONBACKEND_DECLSPEC TRITONSERVER_Error* TRITONBACKEND_ResponseOutput(
     const char* name, const TRITONSERVER_DataType datatype,
     const int64_t* shape, const uint32_t dims_count);
 
+/// Query the response allocator for the preferred memory type and memory
```

Comment:
This needs to be TRITONBACKEND_RequestOutputBufferProperties because the backend may need to determine this information before it generates any of the response objects. Need to document that a return TRITONSERVER_ERROR_UNAVAILABLE indicates that the properties are not available and all other returns indicate an error.